src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -314,6 +322,25 @@ static OnHitEffect enemyPotionEffect(Supplier<MobEffectInstance> effect)
 		return (stack, target, attacker) -> target.addEffect(effect.get());
 	}
 	
+	static OnHitEffect bossBuster()
+	{
+		return (stack, target, attacker) -> {
+			if(attacker instanceof ServerPlayer player && target instanceof UnderlingEntity) {
+				target.hurt(DamageSource.playerAttack(player), target.getMaxHealth() / 20);
+			}
+		};
+	}
+	
+	static OnHitEffect spawnParticles(SimpleParticleType pType, int amount)
+	{
+		return (stack, target, attacker) -> {
+			if(attacker instanceof Player && attacker != null && target != null)
+			{
+				attacker.getLevel().getServer().getLevel(attacker.getLevel().dimension()).sendParticles(pType, target.getX(), target.getY(), target.getZ(), amount, 0.2, 0.5, 0.2, 2);
```

Comment:
Could probably just get the level from that target. See line 265 in this file for an example